lightning/src/ln/channelmanager.rs

```diff
@@ -3011,6 +3034,55 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		Ok(())
 	}
 
+	pub fn fail_intercepted_payment(&self, intercept_id: InterceptId) {
+		let pending_intercept = {
+			let mut pending_intercepts = self.pending_intercepted_payments.lock().unwrap();
+			pending_intercepts.remove(&intercept_id)
+		};
+
+		if let Some(_payment) = pending_intercept {
+			// TODO: what's best way to fail this? in `process_pending_htlc_forwards` it uses that `fail_forward` macro
+			//			 awkward way could be to just use the forward_intercepted_payment code and pick a random scid
+		}
+	}
+
+	pub fn forward_intercepted_payment(&self, intercept_id: InterceptId, scid: u64, amt_to_forward: u64) -> Result<(), APIError> {
+		let pending_intercept = {
+			let mut pending_intercepts = self.pending_intercepted_payments.lock().unwrap();
+			pending_intercepts.remove(&intercept_id)
+		};
+
+		match pending_intercept {
+			None => Err(APIError::APIMisuseError { err: "Payment with that InterceptId not found".to_string() }),
+			Some(payment) => {
+				match payment {
+					HTLCForwardInfo::AddHTLC { prev_short_channel_id, prev_htlc_id, forward_info, prev_funding_outpoint } => {
+
+						let routing = match forward_info.routing {
+								PendingHTLCRouting::Forward { onion_packet, .. } => {
+									PendingHTLCRouting::Forward { onion_packet, short_channel_id: scid }
+								},
+								_ => forward_info.routing
+							};
+
+							let pending_htlc_info = PendingHTLCInfo {
+								amt_to_forward,
+								routing,
+								..forward_info
+							};
+
+							let mut per_source_pending_forward = vec![(prev_short_channel_id, prev_funding_outpoint, vec![(pending_htlc_info, prev_htlc_id)])];
+
+							self.forward_htlcs(&mut per_source_pending_forward);
+
+							Ok(())
+					},
+					_ => Err(APIError::APIMisuseError { err: "impossible".to_string() })
```

Comment:
Not sure how to handle this exactly.  It should be impossible to get into this state.

Reply:
with a) i assume when you say "store that struct here" you mean in pending_intercepted_payments?

b) does kind of suck but I guess would you accept a PR that just did that?

Reply:
went for a